### PR TITLE
Overlay plots in tf asymmetry mode

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

In Muon Analysis TF asymmetry mode the normalised data was being plotted on the same graph as the un-normalised fit. This leads to the fit and the data appearing to be offset. This PR adds a normalised fit spectra to the output of CalculateMuonAsymmetry and updates the plotting in the GUI to plot this normalised fit if it is available.

**To test:**

  * In Muon Analysis 2 load MUSR62260, go to the fitting tab perform the following fits and check that they produce sensible plots.
  * Single fit in non tf mode
  * Single fit in tf mode
  * A simultaneous fit of all the groups in non tf mode
  * A simultaneous fit of all the groups in tf mode

Fixes #26583 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Muon/src/CalculateMuonAsymmetry.cpp [hunk 1]
```diff
@@ -203,11 +203,52 @@ void CalculateMuonAsymmetry::exec() {
...
+    } else {
+      fitWorkspaceActual =
+          boost::dynamic_pointer_cast<API::MatrixWorkspace>(fitWorkspace);
+    }
+    API::IAlgorithm_sptr extractSpectra =
+        createChildAlgorithm("ExtractSingleSpectrum");
+    API::IAlgorithm_sptr appendSpectra = createChildAlgorithm("AppendSpectra");
+
+    extractSpectra->setProperty("InputWorkspace", fitWorkspaceActual);
+    extractSpectra->setProperty("WorkspaceIndex", 2);
```
Comment (AnthonyLim23) [inline Framework/Muon/src/CalculateMuonAsymmetry.cpp]:
```suggestion
    extractSpectra->setProperty("WorkspaceIndex", 1);
```

File: Framework/Muon/src/CalculateMuonAsymmetry.cpp [hunk 2]
```diff
@@ -203,11 +203,52 @@ void CalculateMuonAsymmetry::exec() {
...
     normWS->mutableY(0) = ws->y(0) / norms[j];
     normWS->mutableY(0) -= 1.0;
     normWS->mutableE(0) = ws->e(0) / norms[j];
 
+    unnormalisedFit->mutableY(0) = fitWorkspaceActual->y(1) / norms[j];
+    unnormalisedFit->mutableY(0) -= 1.0;
+    unnormalisedFit->mutableE(0) = fitWorkspaceActual->e(1) / norms[j];
+
+    appendSpectra->setProperty("InputWorkspace1", fitWorkspaceActual);
+    appendSpectra->setProperty("InputWorkspace2", unnormalisedFit);
```
Comment (AnthonyLim23) [inline Framework/Muon/src/CalculateMuonAsymmetry.cpp]: a lot of this looks like it should be in a function

File: Framework/Muon/src/CalculateMuonAsymmetry.cpp [hunk 3]
```diff
@@ -203,6 +203,26 @@ void CalculateMuonAsymmetry::exec() {
     API::MatrixWorkspace_sptr normWS =
         API::AnalysisDataService::Instance().retrieveWS<API::MatrixWorkspace>(
             wsNames[j]);
+
+    normWS->mutableY(0) = ws->y(0) / norms[j];
```
Comment (AnthonyLim23) [inline Framework/Muon/src/CalculateMuonAsymmetry.cpp]: I think we should split off the workspace transform into its own function and we can call it twice. Then when we fix the missing constant we just need to edit one place.

File: Framework/Muon/src/CalculateMuonAsymmetry.cpp [hunk 4]
```diff
@@ -220,13 +221,63 @@ void CalculateMuonAsymmetry::exec() {
   }
 }
 
+/**
+ * Adds a normalised fit asthe last spectra in the fitted workspace.
```
Comment (AnthonyLim23) [inline Framework/Muon/src/CalculateMuonAsymmetry.cpp]:
```suggestion
 * Adds a normalised fit as the last spectra in the fitted workspace.
```

**Review by AnthonyLim23 — CHANGES_REQUESTED** [review]
code looks good, need to test it

**Review by AnthonyLim23 — DISMISSED** [review]

**Comment by martyngigg** [comment]
@AnthonyLim23 If you're happy with this still I can merge it once the builds pass.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**